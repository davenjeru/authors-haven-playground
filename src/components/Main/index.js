import React from 'react';
import PropTypes from 'prop-types';
import './Main.css';

const Main = ({ children, ...rest }) => (
  <main {...rest} role="main">
    {children}
  </main>
);

Main.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.oneOfType(
        [PropTypes.bool, PropTypes.shape(React.Component)],

      )),
    PropTypes.shape(),
  ]),
};

Main.defaultProps = {
  children: <React.Fragment />,
};

export default Main;
