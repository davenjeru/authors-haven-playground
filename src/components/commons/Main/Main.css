main {
    margin-top: 3.55em;
    min-height: 100vh;
}
