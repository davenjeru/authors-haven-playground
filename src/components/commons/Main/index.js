import React from 'react';
import PropTypes from 'prop-types';
import './Main.css';

const Main = ({ children, ...rest }) => (
  <main {...rest} role="main">
    {children}
  </main>
);

Main.propTypes = {
  children: PropTypes.shape(),
};

Main.defaultProps = {
  children: <React.Fragment />,
};

export default Main;
