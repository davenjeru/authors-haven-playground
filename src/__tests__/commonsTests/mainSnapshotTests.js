import renderer from 'react-test-renderer';
import React from 'react';
import Main from '../../components/commons/Main';

describe('The Main element', () => {
  it('should not regress', () => {
    const tree = renderer.create(<Main />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
