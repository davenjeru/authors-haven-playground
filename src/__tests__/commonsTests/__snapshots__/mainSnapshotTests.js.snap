// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Main element should not regress 1`] = `
<main
  role="main"
/>
`;
